animations-dbus/animations-dbus-server.c

```diff
@@ -0,0 +1,624 @@
+ * animations_dbus_server_create_animation_manager:
+ * @server: An #AnimationsDbusServer.
+ * @error: A #GError.
+ *
+ * Create a #AnimationsDbusServerAnimationManager for use by the
+ * server side. The animation manager will appear on the bus but
+ * will be managed by the caller (for instance, there is no bus name
+ * that is being watched for the animation manager to be destroyed).
+ *
+ * Returns: (transfer full): A #AnimationsDbusServerAnimationManager
+ * or %NULL with @error set in case of an error.
+ */
+AnimationsDbusServerAnimationManager *
+animations_dbus_server_create_animation_manager (AnimationsDbusServer  *server,
+                                                 GError               **error)
+{
+  AnimationsDbusServerPrivate *priv = animations_dbus_server_get_instance_private (server);
+  g_autoptr(AnimationsDbusServerAnimationManager) server_animation_manager =
+    animations_dbus_server_animation_manager_new (priv->connection,
+                                                  server,
+                                                  priv->effect_factory);
+
+  /* Need to be careful to use preincrement here as we will be
+   * doing a reverse lookup of this later and we can't have 0
+   * as 0 == NULL, which would indicate that an id was not found
+   * for a given client name */
+  unsigned int allocated_id = ++priv->animation_manager_serial;
+  g_autofree char *object_path = g_strdup_printf (LIBANIMATION_ANIMATION_MANAGER_OBJECT_PATH_TEMPLATE,
+                                                  allocated_id);
+
+  if (!animations_dbus_server_animation_manager_export (server_animation_manager,
+                                                        object_path,
+                                                        error))
+    return NULL;
+
+  /* Because the caller could be a user of the server library and not
+   * the server library itself
+   * (from on_animation_connection_manager_register_client), we don't
+   * register the newly created AnimationsDbusServerAnimationManager in
+   * the hash table, since it is up to the caller to keep track of its
+   * lifetime. */
+  return g_steal_pointer (&server_animation_manager);
+}
+
+static gboolean
+find_matching_value (gpointer key G_GNUC_UNUSED,
+                     gpointer value,
+                     gpointer user_data)
+{
+  return g_strcmp0 (value, user_data) == 0;
+}
+
+static void
+on_animation_manager_owner_name_lost (GDBusConnection *connection G_GNUC_UNUSED,
+                                      const char      *name,
+                                      gpointer         user_data)
+{
+  AnimationsDbusServer *server = user_data;
+  AnimationsDbusServerPrivate *priv = animations_dbus_server_get_instance_private (server);
+  guint watch_id = 0;
+
+  if (g_hash_table_lookup_extended (priv->client_name_watches,
+                                    name,
+                                    NULL,
+                                    (gpointer *) &watch_id))
```

Comment:
I think in the case of `g_hash_table_lookup_extended`, it wants a `gpointer *`. That said, maybe I can use `&GUINT_TO_POINTER (watch_id)`

Reply:
This will have to be:
```
gpointer watch_id_ptr;
if (g_hash_table_lookup_extended (priv->client_name_watches, name, NULL, &watch_id_ptr))
  {
    gint watch_id = GPOINTER_TO_INT (watch_id_ptr);
    …
  }
```